packages/socket.io/lib/namespace.ts

```diff
@@ -322,6 +322,9 @@ export class Namespace<
     debug("adding socket to nsp %s", this.name);
     const socket = await this._createSocket(client, auth);
 
+    // track socket
+    this.sockets.set(socket.id, socket);
```

Comment:
Hmm, I'm not sure about this change, since the `sockets` map currently contains only sockets that are actually connected (not in the process of connection).